tools/bqtag/bqtag/bqtag.py

```diff
@@ -0,0 +1,800 @@
+                                        p=self.catalog_project,
+                                        l=self.location.lower()
+                                    )
+                                )
+        except (MethodNotImplemented, NotFound) as e:
+            LOGGER.error(
+                            "Could not fetch Policy Tags. \
+                            API call to Google failed. Error received: %s",
+                            str(e)
+                        )
+            LOGGER.error(traceback.format_exc())
+            return False
+
+        for taxonomy in taxonomies:
+            if taxonomy.display_name == self.taxonomy:
+                self.taxonomy_id = taxonomy.name
+                break
+
+        if not self.taxonomy_id:
+            LOGGER.error(
+                            "Could not determine Taxonomy ID for taxonomy: %s. \
+                            Please check if Taxonomy name, Data Catalog Project \
+                            and Location are correct.",
+                            self.taxonomy
+                        )
+            return False
+
+
+        LOGGER.debug("Downloading Policy Tags from Taxonomy.")
+
+        policy_tags = self.dc.list_policy_tags(parent=self.taxonomy_id)
+
+        for policy_tag in policy_tags:
+            self.policy_tags[policy_tag.display_name] = policy_tag.name
+
+        self.policy_tags_rev = { k:v for v,k in self.policy_tags.items()}
+
+        LOGGER.info(
+                        "Policy Tags downlaoded from Taxonomy. Total tags: %s",
+                        str(len(self.policy_tags)
+                    )
+                )
+
+        return True
+
+
+    # create dataset
+    def create_dataset(self) -> bool:
+
+        """
+        Create a new BQ Dataset
+        :return True if success and False if Failure
+        """
+
+        dataset = bigquery.Dataset(".".join([self.bq_project, self.dataset]))
+        dataset.location = self.location
+
+        try:
+            dataset = self.bq.create_dataset(dataset, timeout=30)
+        except (MethodNotImplemented, NotFound) as e:
+            LOGGER.error("Could not create dataset: %s", str(e))
+            LOGGER.error(traceback.format_exc())
+            return False
+
+        LOGGER.info("Dataset created successfully: %s.%s.",
+                    self.bq_project,
+                    self.dataset
+                )
+        return True
+
+
+    # Create a table with tags
+    def create_table(self,
+                     table_name: str,
+                     table_schema: str,
+                     table_tag_map: dict = None) -> str:
+        """
+        Create a new Tagged table
+        :param table_name: Name of the table to create
+        :param table_schema: Schema of the table to create
+        :param table_tag_map: Mapping of Tags to Columns
+        :return json containing schema of table created
+        """
+
```

Comment:
Remove blank line

Reply:
def create_table(self,
                     table_name: str,
                     table_schema: str,
                     table_tag_map: dict = None) -> str:
        """
        Create a new Tagged table.
        :param table_name: Name of the table to create
        :param table_schema: Schema of the table to create
        :param table_tag_map: Mapping of Tags to Columns
        :return: json containing schema of table created
        """
        tagged_schema = self._process_schema(table_schema=table_schema,